### PR TITLE
Fixing SCIP errors

### LINKED ISSUE #2545: Pyomo 6.4 can't use scip 8.0.1
As title, when I use scip to solve my problem, it can't start to solve problem:
image

### PR DESCRIPTION
**Pull request by whart222**
## Summary/Motivation:
Fixing SCIP runtime errors.

## Changes proposed in this PR:
- Miscellaneous changes to documentation and error messages
- Use the problem sense to appropriately set the value of lower and upper bound values after optimization.
- Confirmed that SCIP worked on all test problems.

### Legal Acknowledgement

By contributing to this software project, I have read the [contribution guide](https://pyomo.readthedocs.io/en/stable/contribution_guide.html) and agree to the following terms and conditions for my contribution:

1. I agree my contributions are submitted under the BSD license.
2. I represent I am authorized to make the contributions and grant the license. If my employer has rights to intellectual property that includes these contributions, I represent that I have received permission to make contributions and grant the required license on behalf of that employer.

### CODE REVIEW

**Comment by whart222** [comment]
@mrmundt Checkout this PR.  It was obvious how to add SCIP where GLPK was being used.  But is that only impacts Windows tests.  Is there an obvious place to add SCIP tests with Linux?

**Comment by mrmundt** [comment]
@whart222 - Added!

File: .github/workflows/test_branches.yml [hunk 1]
```diff
@@ -50,15 +50,15 @@ jobs:
...
+          PACKAGES: glpk scip
 
         - os: ubuntu-latest
           python: 3.9
           other: /conda
           skip_doctest: 1
           TARGET: linux
           PYENV: conda
-          PACKAGES:
+          PACKAGES: scip
```
Comment (jsiirola) [inline .github/workflows/test_branches.yml:61]: The standard place to put packages that should only be installed in conta environments is line 311:
```
for PKG in cyipopt pymumps; do
```

Comment (whart222) [reply .github/workflows/test_branches.yml:61]: Thx.  I made that change.

**Comment by whart222** [comment]
@jsiirola @mrmundt @blnicho Can you help me diagnose the test failure for Linux?  This looks like a failure that is not triggered by my changes.  Yes?

**Comment by emma58** [comment]
@whart222 the linux and osx 3.11 failure in the appsi tests is not your fault--those are happening everywhere at the moment. I'm not sure about Jenkins though?

**Comment by whart222** [comment]
I've revised this PR to make it clear that this seems to be "done".  I thought I was going to work on SCIP execution failures, but I really didn't see any.  I did offline tests on a real application, again without seeing any issues using SCIP.

File: pyomo/solvers/plugins/solvers/SCIPAMPL.py [hunk 2]
```diff
@@ -396,6 +396,10 @@ def _postsolve(self):
             if len(results.solution) > 0:
                 results.solution(0).status = \
                     SolutionStatus.optimal
+            if results.problem.sense == ProblemSense.minimize:
+                results.problem.lower_bound = results.solver.primal_bound
+            else:
+                results.problem.upper_bound = results.solver.primal_bound
```
Comment (emma58) [inline pyomo/solvers/plugins/solvers/SCIPAMPL.py:402]: @whart222 I think the Jenkins failures are related to these lines. There are failures in the pickle tests and in the writer tests, all complaining that `primal_bound` is an unknown attribute on the `SolverInformation` class. As to why this only comes up on Jenkins and not GH actions, maybe @mrmundt or @jsiirola has a theory.

**Review by blnicho — APPROVED** [review]

File: pyomo/solvers/plugins/solvers/SCIPAMPL.py (hunk 2 above)
Comment (mrmundt) [reply pyomo/solvers/plugins/solvers/SCIPAMPL.py:402]: Tests are all passing now.

**Review by mrmundt — APPROVED** [review]

**Pull request merged**